lib/msf/core/post/common.rb

```diff
@@ -251,4 +251,31 @@ def command_exists?(cmd)
     raise "Unable to check if command `#{cmd}' exists"
   end
 
+  def get_processes
+    if session.type == 'meterpreter'
+      return session.sys.process.get_processes
+    end
+    processes = []
+    if session.platform == 'windows'
+      tasklist = cmd_exec('tasklist').split("\n")
+      tasklist.each do |p|
+        properties = p.split
+        process = {}
+        process['name'] = properties[0].to_s
+        process['pid'] = properties[1].to_s
+        processes.push(process)
+      end
+    else
+      ps_aux = cmd_exec('ps aux').split("\n")
```

Comment:
Looks like this one should filter out the first line.

```
msf6 post(windows/gather/checkvm) > irb
[*] Starting IRB shell...
[*] You are in post/windows/gather/checkvm

>> get_processes
=> 
[{"name"=>"COMMAND", "pid"=>"PID"},
 {"name"=>"/usr/lib/systemd/systemd", "pid"=>"1"},
 {"name"=>"kthreadd", "pid"=>"2"},
```

Reply:
yeah, thanks I'll change that.